kernel/kernel-api/src/test/scala/io/delta/kernel/internal/snapshot/LogSegmentSuite.scala

```diff
@@ -234,4 +236,79 @@ class LogSegmentSuite extends AnyFunSuite with MockFileSystemClientUtils {
     // scalastyle:on line.size.limit
     assert(logSegment.toString === expectedToString)
   }
+
+  test("allFilesWithCompactionsReversed -- 3 - 5 in middle") {
```

Comment:
is it possible for log compaction files to overlap with each other? what would happen then?

Reply:
we can have multiple compaction files also right? let's test that? (back to back and separated?)

Reply:
also what about a test where the log compaction file spans the entire version range (i.e. only that should be returned)?

Reply:
should we include checkpoints in some of the test cases?